core/test/PrognosticData_test.cpp

```diff
@@ -1,12 +1,105 @@
 /*!
  * @file PrognosticData_test.cpp
  *
- * @date Mar 1, 2022
+ * @date Jan 27, 2023
  * @author Tim Spain <timothy.spain@nersc.no>
  */
 
-#include "/Users/timpai/nextsim/dg/nextsimdg/core/src/include/PrognosticData.hpp"
+#include "include/PrognosticData.hpp"
+
+#define CATCH_CONFIG_MAIN
+#include <catch2/catch.hpp>
+
+#include "include/PrognosticData.hpp"
+
+#include "include/ConfiguredModule.hpp"
+#include "include/IOceanBoundary.hpp"
+#include "include/ModelComponent.hpp"
+#include "include/Module.hpp"
+#include "include/UnescoFreezing.hpp"
+#include "include/constants.hpp"
+
+#include <sstream>
 
 namespace Nextsim {
 
+TEST_CASE("PrognosticData call order test", "[PrognosticData]")
+{
+    ModelArray::setDimensions(ModelArray::Type::H, { 1, 1 });
+    ModelArray::setDimensions(ModelArray::Type::Z, { 1, 1, 1 });
+
+    std::stringstream config;
+    config << "[Modules]" << std::endl;
+    config << "Nextsim::IAtmosphereBoundary = Nextsim::ConfiguredAtmosphere" << std::endl;
+    config << std::endl;
+    config << "[ConfiguredAtmosphere]" << std::endl;
+    config << "t_air = 3" << std::endl;
+    config << "t_dew = 2" << std::endl;
+    config << "pmsl = 100000" << std::endl;
+    config << "sw_in = 50" << std::endl;
+    config << "lw_in = 330" << std::endl;
+    config << "snow = 0" << std::endl;
+    config << "rainfall = 0" << std::endl;
+    config << "wind_speed = 5" << std::endl;
+
+    std::unique_ptr<std::istream> pcstream(new std::stringstream(config.str()));
+    Configurator::addStream(std::move(pcstream));
+
+    ConfiguredModule::parseConfigurator();
+
+    class OceanData : public IOceanBoundary {
```

Comment:
Hi Tim. Noticed this declaration exists in this form in multiple locations including ConfiguredAtmosphere_test, FiniteElementFluxes_test and BasicIceOceanFlux_test (sometimes multiple times in one file). Could we either make a class in the source (presuming we will need this) or make a common class in the physics/test directory that we can import? I presume the duplication is because this is implemented in a change that is being worked on on another branch but we know is coming soon?

Let's keep maintenance engineering effort as low as possible

Reply:
No, the duplication is because it is supposed to be an ad-hoc class that implements a given `ModelComponent` (here `IOceanBoundary`) and provides whatever data is required by the particular test. And because it is ad-hoc it gets shoved in wherever I thought it was needed.

I would make the argument that breaking the (admittedly) similar code out into a separate class would not necessarily reduce maintenance overhead as all the classes would have to be kept reconciled. Could we at least create a new issue to discuss it. Since it impacts several tests, it should be its own issue, spec and pull request, in my opinion.

Reply:
Yes we can cover this as a follow-up/tidy-up issue. Please create and tag and link here and I'll approve this PR. Nudge me if needed.

Reply:
A relevant issue has been created at #256